### PR TITLE
i18n: make double dollar validation less strict

### PR DESCRIPTION
**Pull request by piotrzarycki**
#10285

### CODE REVIEW

File: lighthouse-core/scripts/i18n/collect-strings.js [hunk 1]
```diff
@@ -342,7 +342,8 @@ function _processPlaceholderDirectIcu(icu, examples) {
  */
 function _ctcSanityChecks(icu) {
   // '$$' i.e. "Double Dollar" is always invalid in ctc.
-  if (icu.message.match(/\$\$/)) {
+  const regexMatch = icu.message.match(/\$([^$]*?)\$/);
```
Comment (exterkamp) [inline lighthouse-core/scripts/i18n/collect-strings.js]: This should be a global match to check for multiple double dollars.

File: lighthouse-core/scripts/i18n/collect-strings.js [hunk 2]
```diff
@@ -342,7 +342,8 @@ function _processPlaceholderDirectIcu(icu, examples) {
...
+  if (regexMatch && !regexMatch[1]) {
```
Comment (exterkamp) [inline lighthouse-core/scripts/i18n/collect-strings.js]: Will need to iterate over all matches.

**Comment by paulirish** [comment]
@piotrzarycki nice work! wondering if you are able to make the updates?

**Comment by piotrzarycki** [comment]
> @piotrzarycki nice work! wondering if you are able to make the updates?

yep, will do it today :)

**Comment by paulirish** [comment]
@piotrzarycki just a friendly reminder :)

**Comment by paulirish** [comment]
@piotrzarycki will you be able to look at this?

**Comment by piotrzarycki** [comment]
> @piotrzarycki will you be able to look at this?

Hi, yes, sorry for delay. For today i gonna make update

**Comment by paulirish** [comment]
@piotrzarycki if you can't finish this up no worries. just let us know. :)

**Comment by piotrzarycki** [comment]
Hi, sorry, i forgot about this task. 
Will update today

**Comment by paulirish** [comment]
@piotrzarycki thanks for sticking with it! we appreciate it. :D

**Comment by piotrzarycki** [comment]
Haha. Yeah i hope next PR-s will be faster

**Pull request merged**